Makefile

```diff
@@ -137,9 +138,23 @@ new-version:
 
 # NOTE: Run make new-version first
 publish: prepublish
+	make setup-publish-npm
 	./node_modules/.bin/lerna publish from-git --require-scripts
+	rm -f .npmrc
 	make clean
 
+setup-publish-npmrc:
+ifeq ("$(CI)", "true")
+ifneq ("$(NPM_TOKEN)", "")
+	echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc
+else
+	echo "Missing NPM_TOKEN env var"
+	exit 1
+endif
+else
+	echo "Skipping .npmrc generation..."
+endif
```

Comment:
I'm pretty sure this works on Linux, but we should try on Mac and Windows first I think.

Reply:
We are already using it in Makefile